html/modules/custom/ncms_ui/src/Plugin/Field/FieldWidget/EntityReferenceBrowserTableWidgetNcms.php

```diff
@@ -163,6 +164,23 @@ private function isSortable() {
     return !$this->isSingleEntitySelect();
   }
 
+  /**
+   * Override buildTableHeaders function to add a custom column.
+   *
+   * @return array
+   *   An array of header titles.
+   */
+  public function buildTableHeaders(): array {
+    $entity_browser = $this->getEntityBrowser();
+    return [
+      '',
+      $this->getFirstColumnHeader(),
+      $entity_browser->id() == 'articles' ? $this->t('Tags') : '',
+      $this->getAdditionalFieldColumnHeader(),
+      $this->getActionColumnHeader(),
+    ];
```

Comment:
Shouldn't this use `array_filter()` too, just like in `buildTableRows()` in line 121?
Just checked and the parent implementation in `Drupal\entity_browser_table\Plugin\Field\FieldWidget\EntityReferenceBrowserTableWidget` also uses `array_filter()` for both. Now I also see where you were coming from with the this snippet that I asked you about in an earlier review pass:
```
function ($header) {
  return $header !== NULL;
}
```
So I'd say here, if entity browser is not 'articles', that line should return NULL instead of '' and array_filter should be used to filter out that single NULL column, because we wouldn't want to have the tags column at all.
The same logic should be applied to `buildTableRows()`.

I'll create a PR against yours I think.

Reply:
Here we go: https://github.com/UN-OCHA/hpc-content-site/pull/334